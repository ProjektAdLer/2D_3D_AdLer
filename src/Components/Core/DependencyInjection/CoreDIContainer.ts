import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
