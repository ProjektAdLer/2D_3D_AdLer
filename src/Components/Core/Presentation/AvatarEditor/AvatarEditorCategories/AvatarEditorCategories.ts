export const OAvatarEditorCategory = {
  HAIR: 0,
  FACE: 1,
  ACCESSOIRE: 2,
  CLOTHING: 3,
  BODY: 4,
} as const;

export type AvatarEditorCategory =
  (typeof OAvatarEditorCategory)[keyof typeof OAvatarEditorCategory];
