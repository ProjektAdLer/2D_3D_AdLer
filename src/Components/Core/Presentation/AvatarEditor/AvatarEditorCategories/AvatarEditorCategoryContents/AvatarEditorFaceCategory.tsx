import RangeSlider from "~ReactComponents/ReactRelated/ReactBaseComponents/RangeSlider";
import AvatarEditorCategoryContentProps from "./AvatarEditorCategoryContentProps";
import faceCategoryIcon from "../../../../../../Assets/icons/face.svg";
import TileGridLayout from "~ReactComponents/GeneralComponents/TileLayout/TileGridLayout";
import { useTranslation } from "react-i18next";

const eyebrowsThumbnails = require.context(
  "../../../../../../Assets/textures/avatar/face/eyebrows",
);
const eyebrowsThumbnailsList = eyebrowsThumbnails
  .keys()
  .map((key) => eyebrowsThumbnails(key));

const eyesThumbnails = require.context(
  "../../../../../../Assets/textures/avatar/face/eyes",
);
const eyesThumbnailsList = eyesThumbnails
  .keys()
  .map((key) => eyesThumbnails(key));

const mouthThumbnails = require.context(
  "../../../../../../Assets/textures/avatar/face/mouths",
);
const mouthThumbnailsList = mouthThumbnails
  .keys()
  .map((key) => mouthThumbnails(key));

const noseThumbnails = require.context(
  "../../../../../../Assets/textures/avatar/face/noses",
);
const noseThumbnailsList = noseThumbnails
  .keys()
  .map((key) => noseThumbnails(key));

export default function AvatarEditorFaceCategory(
  props: AvatarEditorCategoryContentProps,
) {
  const { t: translate } = useTranslation("avatarEditor");

  return (
    <div className="flex flex-col">
      <h1 className="text-2xl font-bold">{translate("eyebrowsTitle")}</h1>
      <TileGridLayout
        tileContents={eyebrowsThumbnailsList.map((image, index) => ({
          id: index,
          image,
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          console.log(id);
        }}
      />

      <h1 className="text-2xl font-bold">{translate("eyesTitle")}</h1>
      <TileGridLayout
        tileContents={eyesThumbnailsList.map((image, index) => ({
          id: index,
          image,
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          console.log(id);
        }}
      />

      <h1 className="text-2xl font-bold">{translate("noseTitle")}</h1>
      <TileGridLayout
        tileContents={noseThumbnailsList.map((image, index) => ({
          id: index,
          image,
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          console.log(id);
        }}
      />

      <h1 className="text-2xl font-bold">{translate("mouthTitle")}</h1>
      <TileGridLayout
        tileContents={mouthThumbnailsList.map((image, index) => ({
          id: index,
          image,
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          console.log(id);
        }}
      />
    </div>
  );
}
