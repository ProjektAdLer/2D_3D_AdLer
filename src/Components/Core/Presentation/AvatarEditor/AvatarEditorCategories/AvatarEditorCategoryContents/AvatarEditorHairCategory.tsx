import TileGridLayout from "~ReactComponents/GeneralComponents/TileLayout/TileGridLayout";
import AvatarEditorCategoryContentProps from "./AvatarEditorCategoryContentProps";
import { useTranslation } from "react-i18next";
import {
  OAvatarBeardModels,
  AvatarHairModels,
  AvatarNoneModel,
  AvatarBeardModels,
  OAvatarHairModels,
} from "../../../../../Core/Domain/AvatarModels/AvatarModelTypes";
import AvatarColorPalette from "../../../../../Core/Domain/AvatarModels/AvatarColorPalette";
import { useState } from "react";
import ColorPickerButton from "~ReactComponents/GeneralComponents/ColorPicker/ColorPickerButton";
import ColorPickerModal from "~ReactComponents/GeneralComponents/ColorPicker/ColorPickerModal";

import noneIcon from "../../../../../../Assets/avatarEditorThumbnails/none_Thumbnail.svg";
import AccordionElement from "~ReactComponents/GeneralComponents/Accordion/AccordionElement";

const noneThumbnail = {
  type: AvatarNoneModel.None,
  image: noneIcon,
};

const hairThumbnails = Object.values(OAvatarHairModels).map<{
  type: AvatarHairModels; // use union type with AvatarNoneModel
  image: string;
}>((type) => ({
  type: type,
  image: require(
    `../../../../../../Assets/avatarEditorThumbnails/hair/hairstyles/${type}.png`,
  ),
}));
hairThumbnails.unshift(noneThumbnail);

const beardThumbnails = Object.values(OAvatarBeardModels).map<{
  type: AvatarBeardModels; // use union type with AvatarNoneModel
  image: string;
}>((type) => ({
  type: type,
  image: require(
    `../../../../../../Assets/avatarEditorThumbnails/hair/beards/${type}.png`,
  ),
}));
beardThumbnails.unshift(noneThumbnail);

export default function AvatarEditorHairCategory(
  props: AvatarEditorCategoryContentProps,
) {
  const { t: translate } = useTranslation("avatarEditor");
  const [showModal, setShowModal] = useState(false);
  const [hairColor, setHairColor] = useState(AvatarColorPalette[0]);

  const TileGridHairStyles = () => {
    return (
      <TileGridLayout
        tileContents={hairThumbnails.map((thumbnail, index) => ({
          id: index,
          image: thumbnail.image,
          title: translate(thumbnail.type).toString() ?? "",
          active: false,
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          props.controller.onAvatarConfigChanged({
            hair: hairThumbnails[id].type,
          });
        }}
      />
    );
  };

  const TileGridBeards = () => {
    return (
      <TileGridLayout
        tileContents={beardThumbnails.map((thumbnail, index) => ({
          id: index,
          image: thumbnail.image,
          title: translate(thumbnail.type).toString() ?? "",
        }))}
        columns={5}
        mobileColumns={3}
        onTileClick={(id) => {
          props.controller.onAvatarConfigChanged({
            beard: beardThumbnails[id].type,
          });
        }}
      />
    );
  };

  return (
    <div className="flex flex-col">
      <AccordionElement
        header={translate("hairstylesTitle").toString()}
        isOpen={props.viewModel.uiVisiblity.hairMenu.hairstyles}
        content={TileGridHairStyles()}
      />
      <AccordionElement
        header={translate("beardsTitle").toString()}
        isOpen={props.viewModel.uiVisiblity.hairMenu.beards}
        content={TileGridBeards()}
      />
      <div className="w-full p-2 m-2">
        <ColorPickerButton
          className=""
          currentColor={hairColor}
          onClick={() => setShowModal(true)}
        />
        <ColorPickerModal
          title={translate("hairColorTitle")}
          showModal={showModal}
          onClose={() => setShowModal(false)}
          onColorClickFunction={(color) => {
            setHairColor(color);
            props.controller.onAvatarConfigChanged({
              hairColor: color,
            });
          }}
        />
      </div>
    </div>
  );
}
