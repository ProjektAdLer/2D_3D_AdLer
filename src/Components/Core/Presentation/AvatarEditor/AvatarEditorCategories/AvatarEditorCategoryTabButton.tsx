import { AdLerUIComponent } from "src/Components/Core/Types/ReactTypes";
import {
  AvatarEditorCategory,
  OAvatarEditorCategory,
} from "./AvatarEditorCategories";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

import hairCategoryIcon from "../../../../../Assets/icons/hair.svg";
import faceCategoryIcon from "../../../../../Assets/icons/face.svg";
import accessoiresCategoryIcon from "../../../../../Assets/icons/accessoires.svg";
import clothingCategoryIcon from "../../../../../Assets/icons/clothing.svg";
import bodyCategoryIcon from "../../../../../Assets/icons/body.svg";

type AvatarEditorCategoryTabButtonProps = {
  category: AvatarEditorCategory;
  active: boolean;
  toolTip?: string;
  onClick: () => void;
} & AdLerUIComponent;

const AvatarEditorCategoryIcons = {
  [OAvatarEditorCategory.HAIR]: hairCategoryIcon,
  [OAvatarEditorCategory.FACE]: faceCategoryIcon,
  [OAvatarEditorCategory.ACCESSOIRE]: accessoiresCategoryIcon,
  [OAvatarEditorCategory.CLOTHING]: clothingCategoryIcon,
  [OAvatarEditorCategory.BODY]: bodyCategoryIcon,
};

export default function AvatarEditorCategoryTabButton(
  props: AvatarEditorCategoryTabButtonProps,
) {
  return (
    <StyledButton
      data-testid={`avatar-editor-category-tab-${props.category}`}
      className={`${props.active ? "!bg-adleryellow translate-x-[1px] translate-y-[1px] !border-b-2 !border-r-2 !border-transparent " : ""}`}
      color={props.active ? "active" : "default"}
      onClick={props.onClick}
      icon={AvatarEditorCategoryIcons[props.category]}
      title={props.toolTip}
    ></StyledButton>
  );
}
