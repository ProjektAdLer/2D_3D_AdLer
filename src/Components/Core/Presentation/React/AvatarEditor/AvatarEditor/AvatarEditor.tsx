import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <div className="grid order-2 grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg">
          <div className="col-start-1">links</div>
          <div className="col-start-2">rechts</div>
        </div>
        <MenuHeaderBar
          location="editor"
          className="self-center order-1 w-full p-2 font-semibold"
        />
      </div>
    </React.Fragment>
  );
}
