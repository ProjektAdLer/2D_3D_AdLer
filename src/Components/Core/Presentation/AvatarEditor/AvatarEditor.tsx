import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React, { useState } from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";
import AvatarEditorCategoryTabButton from "./AvatarEditorCategories/AvatarEditorCategoryTabButton";
import {
  AvatarEditorCategory,
  OAvatarEditorCategory,
} from "./AvatarEditorCategories/AvatarEditorCategories";
import AvatarEditorHairCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorHairCategory";
import AvatarEditorFaceCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorFaceCategory";
import AvatarEditorPreview from "./AvatarEditorPreview/AvatarEditorPreview";
import LoadingScreen from "~ReactComponents/GeneralComponents/LoadingScreen/LoadingScreen";

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  const [activeTab, setActiveTab] = useState<AvatarEditorCategory>(
    OAvatarEditorCategory.HAIR,
  );

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <MenuHeaderBar
          location="editor"
          className="self-center w-full p-2 font-semibold"
        />
        <div className="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 grow lg:rounded-lg">
          {/* Categories (Left Side) */}

          <div className="flex flex-col portrait:order-2">
            {/* Category Tabs */}
            {/* Category Header */}
            <div className="flex flex-row items-center justify-center">
              <div className="flex flex-col items-center justify-center">
                <div className="text-2xl font-bold text-darkblue">
                  {activeTab === OAvatarEditorCategory.HAIR && "Haare"}
                  {activeTab === OAvatarEditorCategory.FACE && "Gesicht"}
                  {activeTab === OAvatarEditorCategory.ACCESSOIRE &&
                    "Accessoires"}
                  {activeTab === OAvatarEditorCategory.CLOTHING && "Kleidung"}
                </div>
              </div>
            </div>
            <div className="flex flex-row justify-center p-2 space-x-4 ">
              {Object.values(OAvatarEditorCategory)
                .filter((category) => typeof category === "number")
                .map((category) => (
                  <AvatarEditorCategoryTabButton
                    key={category}
                    category={category as AvatarEditorCategory}
                    active={activeTab === category}
                    onClick={() => setActiveTab(category)}
                  />
                ))}
            </div>

            {/* Category Contents */}
            <div className="p-4 max-h-[77vh] portrait:max-h-[30vh] overflow-auto">
              {activeTab === OAvatarEditorCategory.HAIR && (
                <AvatarEditorHairCategory controller={controller} />
              )}
              {activeTab === OAvatarEditorCategory.FACE && (
                <AvatarEditorFaceCategory controller={controller} />
              )}
            </div>
          </div>

          {/* Avatar Preview (Right Side) */}
          <div className="p-4">
            <AvatarEditorPreview className="relative w-full h-full" />
          </div>
        </div>
        <LoadingScreen />
      </div>
    </React.Fragment>
  );
}
