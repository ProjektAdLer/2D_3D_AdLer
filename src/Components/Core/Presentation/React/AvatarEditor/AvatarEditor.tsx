import IAvatarEditorController from "./IAvatarEditorController";
import AvatarEditorViewModel from "./AvatarEditorViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import React, { useState } from "react";
import MenuHeaderBar from "~ReactComponents/GeneralComponents/MenuHeaderBar/MenuHeaderBar";
import AvatarEditorCategoryTabButton from "./AvatarEditorCategories/AvatarEditorCategoryTabButton";
import {
  AvatarEditorCategory,
  OAvatarEditorCategory,
} from "./AvatarEditorCategories/AvatarEditorCategories";
import AvatarEditorHairCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorHairCategory";
import AvatarEditorFaceCategory from "./AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorFaceCategory";

const AvatarEditorCategoryContents = {
  [OAvatarEditorCategory.HAIR]: AvatarEditorHairCategory,
  [OAvatarEditorCategory.FACE]: AvatarEditorFaceCategory,
};

export default function AvatarEditor() {
  const [viewModel, controller] = useBuilder<
    AvatarEditorViewModel,
    IAvatarEditorController
  >(BUILDER_TYPES.IAvatarEditorBuilder);

  const [activeTab, setActiveTab] = useState<AvatarEditorCategory>(
    OAvatarEditorCategory.HAIR,
  );

  if (!viewModel || !controller) return null;

  return (
    <React.Fragment>
      <div className="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden">
        <MenuHeaderBar
          location="editor"
          className="self-center w-full p-2 font-semibold"
        />
        <div className="grid grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg">
          {/* Categories (Left Side) */}
          <div className="portrait:order-2">
            {/* Category Tabs */}
            <div className="flex flex-row justify-center space-x-4 p-4">
              {Object.values(OAvatarEditorCategory)
                .filter((category) => typeof category === "number")
                .map((category) => (
                  <AvatarEditorCategoryTabButton
                    className=""
                    key={category}
                    category={category as AvatarEditorCategory}
                    active={activeTab === category}
                    onClick={() => setActiveTab(category)}
                  />
                ))}
            </div>

            {/* Category Contents */}
            <div className="p-4">
              {AvatarEditorCategoryContents[activeTab]({
                controller,
              })}
            </div>
          </div>

          {/* Avatar Preview (Right Side) */}
          <div className="p-4">AVATAR THUMBNAIL</div>
        </div>
      </div>
    </React.Fragment>
  );
}
