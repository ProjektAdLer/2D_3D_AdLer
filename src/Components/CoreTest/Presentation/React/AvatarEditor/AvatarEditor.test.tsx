import React from "react";
import { act, fireEvent, render } from "@testing-library/react";
import AvatarEditor from "../../../../../Components/Core/Presentation/React/AvatarEditor/AvatarEditor";
import { Provider } from "inversify-react";
import CoreDIContainer from "../../../../Core/DependencyInjection/CoreDIContainer";
import AvatarEditorCategoryTabButton from "../../../../Core/Presentation/React/AvatarEditor/AvatarEditorCategoryTabButton";
import { OAvatarEditorCategory } from "../../../../Core/Presentation/React/AvatarEditor/AvatarEditorCategories";
import AvatarEditorHairCategory from "../../../../Core/Presentation/React/AvatarEditor/AvatarEditorCategoryContents/AvatarEditorHairCategory";
import AvatarEditorFaceCategory from "../../../../Core/Presentation/React/AvatarEditor/AvatarEditorCategoryContents/AvatarEditorFaceCategory";

describe("AvatarEditor", () => {
  test("should renders", () => {
    const { container } = render(
      <Provider container={CoreDIContainer}>
        <AvatarEditor />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  test.each([[OAvatarEditorCategory.HAIR], [OAvatarEditorCategory.FACE]])(
    "should render %s category when tab button is clicked",
    (category) => {
      const result = render(
        <Provider container={CoreDIContainer}>
          <AvatarEditor />
        </Provider>,
      );

      const tabButton = result.getByTestId(
        `avatar-editor-category-tab-${category}`,
      );

      act(() => {
        fireEvent.click(tabButton);
      });

      expect(result.container).toMatchSnapshot();
    },
  );
});
