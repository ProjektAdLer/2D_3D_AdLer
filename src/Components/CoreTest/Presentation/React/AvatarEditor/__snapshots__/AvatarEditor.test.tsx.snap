// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarEditor should render 0 category when tab button is clicked 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="grid order-2 grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg"
    >
      <div
        class="col-start-1"
      >
        <div
          class="flex flex-row space-x-4 p-4"
        >
          <button
            class="grow bg-adlerbggradientfrom"
            data-testid="avatar-editor-category-tab-0"
          >
            hair
          </button>
          <button
            class="grow bg-adlerbggradientto"
            data-testid="avatar-editor-category-tab-1"
          >
            face
          </button>
        </div>
        <div
          class="p-4"
        >
          <div>
            HAARE
          </div>
        </div>
      </div>
      <div
        class="col-start-2"
      >
        AVATAR THUMBNAIL
      </div>
    </div>
    <div
      class="self-center order-1 w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarEditor should render 1 category when tab button is clicked 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="grid order-2 grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg"
    >
      <div
        class="col-start-1"
      >
        <div
          class="flex flex-row space-x-4 p-4"
        >
          <button
            class="grow bg-adlerbggradientto"
            data-testid="avatar-editor-category-tab-0"
          >
            hair
          </button>
          <button
            class="grow bg-adlerbggradientfrom"
            data-testid="avatar-editor-category-tab-1"
          >
            face
          </button>
        </div>
        <div
          class="p-4"
        >
          <div>
            GESICHT
          </div>
        </div>
      </div>
      <div
        class="col-start-2"
      >
        AVATAR THUMBNAIL
      </div>
    </div>
    <div
      class="self-center order-1 w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarEditor should renders 1`] = `
<div>
  <div
    class="flex flex-col h-[100svh] bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto overflow-hidden"
  >
    <div
      class="grid order-2 grid-cols-2 grid-rows-1 portrait:grid-cols-1 portrait:grid-rows-2 portrait:gap-4 grow lg:rounded-lg"
    >
      <div
        class="col-start-1"
      >
        <div
          class="flex flex-row space-x-4 p-4"
        >
          <button
            class="grow bg-adlerbggradientfrom"
            data-testid="avatar-editor-category-tab-0"
          >
            hair
          </button>
          <button
            class="grow bg-adlerbggradientto"
            data-testid="avatar-editor-category-tab-1"
          >
            face
          </button>
        </div>
        <div
          class="p-4"
        >
          <div>
            HAARE
          </div>
        </div>
      </div>
      <div
        class="col-start-2"
      >
        AVATAR THUMBNAIL
      </div>
    </div>
    <div
      class="self-center order-1 w-full p-2 font-semibold flex place-content-stretch"
    >
      <div
        class="flex items-center w-1/2 place-content-stretch justify-self-start "
      >
        <div
          class=""
        >
          <button
            class="mr-4 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          >
            <img
              alt="Home Icon"
              class="w-10 xl:w-12 "
              src="home.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="flex justify-center w-full"
      >
        <div
          class="text-xl truncate lg:text-4xl font-[roboto] flex gap-2 m-2 w-fit h-fit text-adlerdarkblue"
        >
          Avatar-Editor
        </div>
      </div>
      <div
        class="flex justify-end w-1/2 "
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <button
              class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
            >
              <img
                alt="Help Icon"
                class="w-10 xl:w-12"
                src="help.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
